py/objmodule.c

```diff
@@ -34,7 +34,11 @@
 #include "py/runtime.h"
 #include "py/builtin.h"
 
+#ifndef NO_QSTR
+// Only include module definitions when not doing qstr extraction, because the
+// qstr extraction stage also generates this module definition header file.
```

Comment:
@jimmo I'm not sure about this... now that `moduledefs.h` is generated **after** qstr extraction, how do all the module qstr names actually get found?

It works though... I guess because qstr extraction sees all the `MP_REGISTER_MODULE()` definitions now.

Reply:
This now works correctly after removing the 3rd arg of `MP_REGISTER_MODULE()`.